TodoListDaemonWithCert/Program.cs

```diff
@@ -69,65 +76,79 @@ static int Main(string[] args)
             authContext = new AuthenticationContext(authority);
 
             // Initialize the Certificate Credential to be used by ADAL.
-            X509Certificate2 cert = null;
-            X509Store store = new X509Store(StoreLocation.CurrentUser);
-            try
-            {
-                store.Open(OpenFlags.ReadOnly);
-                // Place all certificates in an X509Certificate2Collection object.
-                X509Certificate2Collection certCollection = store.Certificates;
-                // Find unexpired certificates.
-                X509Certificate2Collection currentCerts = certCollection.Find(X509FindType.FindByTimeValid, DateTime.Now, false);
-                // From the collection of unexpired certificates, find the ones with the correct name.
-                X509Certificate2Collection signingCert = currentCerts.Find(X509FindType.FindBySubjectDistinguishedName, certName, false);
-                if (signingCert.Count == 0)
-                {
-                    // No matching certificate found.
-                    return -1;
-                }
-                // Return the first certificate in the collection, has the right name and is current.
-                cert = signingCert.OfType< X509Certificate2>().OrderByDescending(c => c.NotBefore).First();
-            }
-            finally
+            X509Certificate2 cert = ReadCertificateFromStore(certName);
+            if (cert == null)
             {
-                store.Close();
+                Console.WriteLine($"Cannot find active certificate '{certName}' in certificates for current user. Please check configuration");
+                return -1;
             }
 
-            // Then create the certificate credential.
+            // Then create the certificate credential client assertion.
             certCred = new ClientAssertionCertificate(clientId, cert);
 
-            // Call the To Do service 10 times with short delay between calls.
+            // Call the ToDoList service 10 times with short delay between calls.
+            int delay = 1000;
             for (int i = 0; i < 10; i++)
             {
-                Thread.Sleep(3000);
-                PostTodo().Wait();
-                Thread.Sleep(3000);
-                GetTodo().Wait();
+                PostNewTodoItem().Wait();
+                Thread.Sleep(delay);
+                DisplayTodoList().Wait();
+                Thread.Sleep(delay);
             }
-
             return errorCode;
         }
 
-        static async Task PostTodo()
+
+        /// <summary>
+        /// Reads the certificate
+        /// </summary>
+        private static X509Certificate2 ReadCertificateFromStore(string certName)
```

Comment:
>private static X509Certificate2 ReadCertificateFromStore(string certName) [](start = 8, length = 73)

Thought:
could be a function in a generic helper for all samples... then it will make it easy for developers to grab our helper class and modify/extend/use it for their own purposes?